gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/application/use_case/ImportApplicationCRDUseCase.java

```diff
@@ -150,6 +179,76 @@ private void createOrUpdateApplicationMetadata(Input input, BaseApplicationEntit
         }
     }
 
+    private void syncApplicationMembers(Input input, BaseApplicationEntity application) {
+        List<String> existingApplicationMembers = memberQueryService
+            .getMembersByReference(MembershipReferenceType.APPLICATION, application.getId())
+            .stream()
+            .filter(member -> {
+                for (Member.Role role : member.getRoles()) {
+                    if (role.isPrimaryOwner()) {
+                        return false;
+                    }
+                }
+                return true;
+            })
+            .map(Member::getId)
+            .collect(Collectors.toList());
+
+        if (!CollectionUtils.isEmpty(input.crd.getMembers())) {
+            input.crd
+                .getMembers()
+                .forEach(crdMember -> {
+                    if (PRIMARY_OWNER.name().equals(crdMember.getRole())) {
+                        throw new SinglePrimaryOwnerException(APPLICATION);
+                    }
+
+                    applicationCrudService.findById(application.getId(), input.auditInfo.environmentId());
```

Comment:
do we really need to query the DB in a loop there ? I think we already have the app :)